cocos/2d/CCFontFreeType.cpp

```diff
@@ -360,16 +360,17 @@ unsigned char* FontFreeType::getGlyphBitmap(uint64_t theChar, long &outWidth, lo
             if (blendWidth > 0 && blendHeight > 0)
             {
                 long index, index2;
-                blendImage = new (std::nothrow) unsigned char[blendWidth * blendHeight * 2];
-                memset(blendImage, 0, blendWidth * blendHeight * 2);
+                int imageSize = blendWidth * blendHeight * 2;
+                blendImage = new (std::nothrow) unsigned char[imageSize];
+                memset(blendImage, 0, imageSize);
 
                 auto px = outlineMinX - blendImageMinX;
                 auto py = blendImageMaxY - outlineMaxY;
                 for (int x = 0; x < outlineWidth; ++x)
                 {
                     for (int y = 0; y < outlineHeight; ++y)
                     {
-                        index = px + x + ((py + y) * blendWidth);
+                        index = (long)(px + x + ((py + y) * blendWidth));
```

Comment:
does index need to be long?

Reply:
the integer type is enough, i'll fix it later.